src/node/internal/http_url.js

```diff
@@ -0,0 +1,745 @@
+// Copyright Joyent, Inc. and other Node contributors.
```

Comment:
I don't think we should pull this in. What I have been planning is to implement our version of `url.parse(...)` and Node.js' legacy url implementation in terms of `new URL` instead of actually porting over the legacy url code. It will be a slight divergence from Node.js but it'll keep things simpler for us over all so that we do not have to maintain this separate (buggy) legacy url implementation. It also means we do not need the qs and punycode stuff either.